platform/windows/detect.py

```diff
@@ -216,6 +223,8 @@ def configure_msvc(env, manual_msvc_config):
     env.AppendUnique(CPPDEFINES=['NOMINMAX']) # disable bogus min/max WinDef.h macros
     if env["bits"] == "64":
         env.AppendUnique(CPPDEFINES=['_WIN64'])
+    if env['use_llvm']:
+        env.Append(CCFLAGS=['-mssse3', '-msse4.1', '-Wno-c++11-narrowing'])
```

Comment:
What's the reason for these changes? This will prevent running Godot on any platform that doesn't support SSE 4.1.

Reply:
I need those flags to to build it successfully on my PC.

I should admit that I am not too familiar with how those instructions work. It seems like SSSE3 and SSE 4.1 features in those `thirdparty` libraries are enabled automatically, but clang-cl doesn't know that. Do you know how the availability of the instruction sets are detected in Windows?

`-Wno-c++11-narrowing` is set because clang-cl complains something from the Windows header file.

Reply:
At least I can rerun the build without those flags and open an issue to document why those flags are added.

Reply:
It may be for CPU detection and dynamic dispatch in the third party code. MSVC allows you to explicitly use e.g. intrinsics above the architecture specified in the cl.exe command line, whereas gcc (and probably clang) do not afaik. You have to take _great care_ around this area to prevent instructions being compiled in above the level of the CPU (which will cause crashes).

Presumably the non-msvc build already has to get around this issue (there may simply be defines to prevent these instructions being compiled at all in third party libraries).

Reply:
I just realized that clang has a `-march=native` flag like gcc which may solve the problem. Let me do some testing. Python also provide some functionality to get cpu information.

Reply:
> I just realized that clang has a `-march=native` flag like gcc which may solve the problem. Let me do some testing. Python also provide some functionality to get cpu information.

This may be confusing the architecture doing the compile and the target architecture. The build CPU is largely irrelevant (after all we e.g. compile for ARM on x86), the relevant thing here is the minimum architecture for the target platform.

Reply:
We don't pass any of those for Linux builds (`platform/x11/detect.py`), so it shouold not be necessary for Windows. If some thirdparty code fails on this, we need to investigate why, it means it's not doing its detection properly.

Reply:
I see. Let's see if I can understand how those thirdparty libraries (such as `libwebp`) get the information. 

This is one of the error messages:
```
thirdparty\libwebp\src\dsp\alpha_processing_sse41.c(52,26): error: always_inline function '_mm_shuffle_epi8' requires target feature 'ssse3', but would be
      inlined into function 'ExtractAlpha_SSE41' that is compiled without support for 'ssse3'
      const __m128i b0 = _mm_shuffle_epi8(a0, kCstAlpha0);
                         ^
thirdparty\libwebp\src\dsp\alpha_processing_sse41.c(53,26): error: always_inline function '_mm_shuffle_epi8' requires target feature 'ssse3', but would be
      inlined into function 'ExtractAlpha_SSE41' that is compiled without support for 'ssse3'
      const __m128i b1 = _mm_shuffle_epi8(a1, kCstAlpha1);
                         ^
thirdparty\libwebp\src\dsp\alpha_processing_sse41.c(54,26): error: always_inline function '_mm_shuffle_epi8' requires target feature 'ssse3', but would be
      inlined into function 'ExtractAlpha_SSE41' that is compiled without support for 'ssse3'
      const __m128i b2 = _mm_shuffle_epi8(a2, kCstAlpha2);
                         ^
thirdparty\libwebp\src\dsp\alpha_processing_sse41.c(55,26): error: always_inline function '_mm_shuffle_epi8' requires target feature 'ssse3', but would be
      inlined into function 'ExtractAlpha_SSE41' that is compiled without support for 'ssse3'
      const __m128i b3 = _mm_shuffle_epi8(a3, kCstAlpha3);
```

Reply:
The relevant logic is in `dsp/dsp.h`:
```
#if defined(__SSE2__) || defined(WEBP_MSC_SSE2) || defined(WEBP_HAVE_SSE2)
#define WEBP_USE_SSE2
#endif

#if defined(__SSE4_1__) || defined(WEBP_MSC_SSE41) || defined(WEBP_HAVE_SSE41)
#define WEBP_USE_SSE41
#endif
```

I assume you're having this issue when using Clang with MSVC?
libwebp is written with the assumption that MSVC means that it can do what @lawnjelly described in https://github.com/godotengine/godot/pull/36580#discussion_r385167821, but evidently that's not possible when using Clang instead of MS VC++.